crmsh/corosync.py

```diff
@@ -669,215 +649,370 @@ def remove_qdevice_db(self):
         parallax.parallax_call(node_list, cmd, self.askpass)
 
 
-def corosync_tokenizer(stream):
-    """Parses the corosync config file into a token stream"""
-    section_re = re.compile(r'(\w+)\s*{')
-    value_re = re.compile(r'(\w+):\s*([\S ]+)')
-    path = []
-    while stream:
-        stream = stream.lstrip()
-        if not stream:
-            break
-        if stream[0] == '#':
-            end = stream.find('\n')
-            t = Token(_tCOMMENT, [], stream[:end])
-            stream = stream[end:]
-            yield t
-            continue
-        if stream[0] == '}':
-            t = Token(_tEND, [])
-            stream = stream[1:]
-            yield t
-            path = path[:-1]
-            continue
-        m = section_re.match(stream)
-        if m:
-            path.append(m.group(1))
-            t = Token(_tBEGIN, path, m.group(1))
-            stream = stream[m.end():]
-            yield t
-            continue
-        m = value_re.match(stream)
-        if m:
-            t = Token(_tVALUE, path + [m.group(1)], m.group(1), m.group(2))
-            stream = stream[m.end():]
-            yield t
-            continue
-        raise ValueError("Parse error at [..%s..]" % (stream[:16]))
+class Parser(object):
+    """
+    Class to parse config file which syntax is similar with corosync.conf
+    Parse corosync.conf by default
+    """
+    INDENT = 4
 
+    def __init__(self, config_file=None):
+        """
+        Init function
+        """
+        self._config_file = config_file if config_file else conf()
+        self._config_dict = {}
+        self._config_list = []
 
-def make_section(path, contents=None):
-    "Create a token sequence representing a section"
-    if not contents:
-        contents = []
-    sp = path.split('.')
-    name = sp[-1]
-    for t in contents:
-        if t.path and not t.path.startswith(path):
-            raise ValueError("%s (%s) not in path %s" % (t.path, t.key, path))
-    return [Token(_tBEGIN, sp, name)] + contents + [Token(_tEND, [])]
+    @staticmethod
```

Comment:
In my daily work, a staticmethod in Python is not really useful. In most cases you can move it out of the class and use the method as a dedicated, separate function. The only reason to use it is for organizational purposes.

If you want to keep it, you should think about a classmethod. You should also change the calling of the method, for example:

```python
# in crmsh/corosync.py: 
 return inner_key in self._inner_section_name_list
```

to this:

```python
# in crmsh/corosync.py: 
 return inner_key in Parser._inner_section_name_list
```

Reply:
Yeah, the big reason for using staticmethod here is easy to read/find under a related class